lib/src/main/java/graphql/nadel/engine/transform/NadelTypeRenameResultTransform.kt

```diff
@@ -77,13 +77,16 @@ internal class NadelTypeRenameResultTransform : NadelTransform<State> {
                 underlyingTypeName = underlyingTypeName,
             )
 
-            val typeName: String = if (overallField.objectTypeNames.contains(overallTypeName)) {
+            val typeName: String = if (
+                !executionContext.hints.sharedTypeRenames(service) ||
+                overallField.objectTypeNames.contains(overallTypeName)
+            ) {
```

Comment:
so, if hints are off, we will always use the old behaviour, just get the `overallTypeName`